src/searchbar/SearchBar-android.js

```diff
@@ -54,14 +54,15 @@ class SearchBar extends Component {
     super(props);
     this.state = {
       hasFocus: false,
-      isEmpty: true,
+      isEmpty: !props.value,
```

Comment:
Nice catch ! Maybe also check for `defaultValue` which is a react native `TextInput` props